File: app/src/main/java/com/openclassrooms/realestatemanager/data/local/entities/PropertyWithImages.java
package com.openclassrooms.realestatemanager.data.local.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;


public class PropertyWithImages {
    @Embedded public SingleProperty mSingleProperty;
    @Relation(
            parentColumn = "pid",
            entityColumn = "property_id"
    )
    public List<ImageOfProperty> ImagesOfProperty;
}

File: app/src/main/java/com/openclassrooms/realestatemanager/data/local/entities/PropertyWithImages.kt
package com.openclassrooms.realestatemanager.data.local.entities

import androidx.room.Embedded
import androidx.room.Relation

class PropertyWithImages {
    @JvmField
    @Embedded
    var mSingleProperty: SingleProperty? = null

    @JvmField
    @Relation(parentColumn = "pid", entityColumn = "property_id")
    var ImagesOfProperty: List<ImageOfProperty>? = null
}